File: pinyougou-manager-web/src/main/webapp/js/service/specificationService.js
app.service("specificationService",function ($http) {
    //1.带分页的条件查询
    this.search = (page,pageSize,specName)=>{
       return $http.get("/specification/search.do?page="+page+"&pageSize="+pageSize+"&specName="+specName);
    };
    //2. 保存方法
    this.save = (url,entity) =>{
        return $http.post(url,entity);
    };

    //3.修改规格，查询原来的规格信息
    this.findOne = id =>{
        return $http.get("/specification/findOne.do?id="+id);
    };

    //4.删除方法
    this.dele = ids =>{
        return $http.get("/specification/delete.do?ids="+ids);
    };
    //5.查询所有
    this.findAll = ()=>{
        return $http.get("/specification/findAll.do");
    }
});

File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/SpecificationService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.group.Specification;
import com.pinyougou.pojo.PageResult;
import com.pinyougou.pojo.TbSpecification;

import java.util.List;
import java.util.Map;


/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface SpecificationService {


    //分页查询，模糊查询
    PageResult search(int page, int pageSize, String key);

    //添加规格
    void add(Specification specification);

    //修改规格信息
    void update(Specification specification);

    //找出当前规格信息
    Specification findOne(Long id);

    List<Map> findAll();

    void delete(Long[] ids);
}
